### DISCUSSION TITLE
Heroku Django Deployment Error

### DISCUSSION THREAD

**Discussion by Sophyia7** [planetscale/discussion#241] [Q&A] [answered]
I tried deploying my app on Heroku, and it returned an error that says `django_psdb_engine` is not found. How do I resolve this? 

This is the error message:
![Screenshot 2022-07-25 145329](https://user-images.githubusercontent.com/66218403/180825000-24a96f6e-b2bc-46d3-9987-4b25dcd2bed4.png)

**Comment by jgreet** [comment]
Perhaps it just can't find the files to import? Did you check in the `django_psdb_engine` files in the root of your project?

You can see them in the example app here:
https://github.com/planetscale/django-example/tree/main/mysite/django_psdb_engine

**Reply by Sophyia7** [reply]
I cloned and imported it. It didn't work.

**Comment by fooforge** [comment, marked as answer]
I'm fairly certain this is an issue with how your project is set up. You either haven't checked in the `custom_db_backends` and `django_psdb_engine` directories or you haven't deployed them to Heroku, or it's something else entirely.

I have taken the time today to alter our [django-example](https://github.com/planetscale/django-example) so that I could deploy it to Heroku. You can find the working code and commands I have used in [django-heroku-example](https://github.com/fooforge/django-heroku-example). The best course of action here would be for you to take a look and compare your repository with mine to learn what needs tweaking.

As an aside, for a production app, you likely want to use Heroku's Config Vars feature instead of adding your database credentials directly to your `settings.py` like I did in the example. Heroku also has some more [information on Django in their help center](https://devcenter.heroku.com/categories/working-with-django) which I would recommend reading through.

**Reply by Sophyia7** [reply]
Thanks. I will check this out.
